activestorage/active.py

```diff
@@ -20,36 +22,115 @@ class Active:
     Version 2 will add methods for actual active storage.
 
     """
-    def __init__(self,uri,*args,**kwargs):
-        """ Instantiate in the same way as normal """
+    def __new__(cls, *args, **kwrags):
+        """Store reduction methods."""
+        instance = super().__new__(cls)
+        instance._methods = {
+            "min": np.min,
+            "max": np.max,
+            "sum": np.sum,
+            # For the unweighted mean we calulate the sum and divide
+            # by the number of non-missing elements
+            "mean": np.sum,
+        }
+        return instance
+
+    def __init__(self, uri, ncvar=None):
+        """Instantiate in the same way as normal.
+
+        :Parameters:
+
+            ncvar: `str`, optional
+                The netCDF variable name of the data. May also be set
+                with the `ncvar` attribute.
+
+        """
         # Assume NetCDF4 for now
-        self.file = Dataset(uri)
-        self._version = 1
         self.uri = uri
-        self.method = None
+        self.ncvar = ncvar
         self.zds = None
 
-    def __getitem__(self, *args):
+        self._version = 1
+        self._components = False
+        self._method = None
+
+    def __getitem__(self, index):
         """ 
         Provides support for a standard get item.
         """
         # In version one this is done by explicitly looping over each chunk in the file
         # and returning the requested slice ourselves. In version 2, we can pass this
         # through to the default method.
-        if self.method is not None and self._version != 2:
-            raise ValueError(f'Cannot apply active storage with this version ({self._version}) of the getitem method')
-        if self._version == 0:
-            return self.file.__getitem__(*args)
+        ncvar = self.ncvar
+        if ncvar is None:
+            raise ValueError("Must set a netCDF variable name to slice")
+
+        if self.method is None and self._version == 0:
+            # No active operation
+            nc = Dataset(self.uri)
+            data = nc[ncvar][index]
+            nc.close()
+            return data
         elif self._version == 1:
-            return self._via_kerchunk(*args)
+            return self._via_kerchunk(index)
```

Comment:
Why index and not *args?

Reply:
Hi Bryan, this is because the parameters are already tuple-ised by the getitem magic.
```python
>>> class A:
...     def __getitem__(self, index):  # No *
...         print(index)
... 
>>> a = A()
>>> a[1, 2, 3]
(1, 2, 3)
```
```python
>>> class B:
...     def __getitem__(self, *index):  # with *
...         print(index)
... 
>>> 
>>> b = B()
>>> a[1, 2, 3]
((1, 2, 3),)
```
It is usual to not star-ise these arguments  (e.g. https://numpy.org/doc/stable/reference/generated/numpy.ndarray.__getitem__.html, https://github.com/dask/dask/blob/2022.9.2/dask/array/core.py#L1924)

Reply:
Thanks, I'm slightly surprised that hadn't caused problems further down the stack, but yes, no start is better!

Reply:
it's not causing any issues since the star is still interpreted correctly, Bryan